project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java

```diff
@@ -177,4 +188,44 @@ public void onRadioButtonClicked(View view) {
         return keywordList;
     }
 
+    /**
+     * This method check if user entered valid input
+     * The result will return to decide if the experiment can be create
+     * @param description
+     * @param name
+     * @param keywords
+     * @return
+     */
+    public boolean checkEnterValid(String description, String name, ArrayList<String> keywords){
+        boolean checkIfValid=true;
+        if(description==null||description.length()==0){
+            checkIfValid = false;
+        }
+        if(name==null||name.length()==0){
+            checkIfValid = false;
+        }
+        if(keywords.isEmpty()){
+            checkIfValid = false;
+        }
+        return checkIfValid;
+    }
+
+    /**
+     * This method is to check whether the checkbox got selected
+     * return true if selected, false for not
+     * @return
+     */
+    public boolean checkBoxSelecte(){
```

Comment:
I recently added in the functionality for the "Subscribe to experiment?" checkbox as well, I'm not sure if it's necessary but could be worth adding that the check for that in here now that it actually does something? It's an optional input though, so I'm not sure it even needs to be checked.

Reply:
I think all the checkboxes are optional? This method may not be needed then.